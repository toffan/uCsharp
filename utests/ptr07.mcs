int* malloc(int size) {
    int* addr;
    asm {
        LOAD %size
        SUBR Malloc
        STORE %addr
    }
    return addr;
}

typedef struct {int x; int y;} Pt;
int main() {
    Pt* p;
    p = (Pt*) malloc(2);

    return 0;
}
