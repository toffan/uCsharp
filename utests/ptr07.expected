JUMP __malloc
_malloc:
    PUSH 1
	LOAD (1) -1[LB]
	SUBR Malloc
	STORE (1) 3[LB]
    LOAD (1) 3[LB]
    RETURN (1) 1
    RETURN (0) 1
__malloc:

JUMP __main
_main:
    PUSH 1
    LOADL 2
    CALL (SB) _malloc
    LOAD (1) -1[ST]
    LOADL 3
    STOREI (1)
    POP (0) 1

    LOADL 0
    RETURN (1) 0
    RETURN (0) 0
__main:

    CALL (SB) _main
    HALT
